drivers/i2c/i2c_gecko.c

```diff
@@ -197,68 +197,33 @@ static const struct i2c_driver_api i2c_gecko_driver_api = {
 	.transfer = i2c_gecko_transfer,
 };
 
-#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
-
-#define PIN_I2C_0_SDA {DT_INST_PROP_BY_IDX(0, location_sda, 1), \
-		DT_INST_PROP_BY_IDX(0, location_sda, 2), gpioModeWiredAnd, 1}
-#define PIN_I2C_0_SCL {DT_INST_PROP_BY_IDX(0, location_scl, 1), \
-		DT_INST_PROP_BY_IDX(0, location_scl, 2), gpioModeWiredAnd, 1}
-
-static const struct i2c_gecko_config i2c_gecko_config_0 = {
-	.base = (I2C_TypeDef *)DT_INST_REG_ADDR(0),
-	.clock = cmuClock_I2C0,
-	.pin_sda = PIN_I2C_0_SDA,
-	.pin_scl = PIN_I2C_0_SCL,
 #ifdef CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION
-	.loc_sda = DT_INST_PROP_BY_IDX(0, location_sda, 0),
-	.loc_scl = DT_INST_PROP_BY_IDX(0, location_scl, 0),
+#define I2C_LOC_DATA(idx) \
+	.loc_sda = DT_INST_PROP_BY_IDX(idx, location_sda, 0), \
+	.loc_scl = DT_INST_PROP_BY_IDX(idx, location_scl, 0)
 #else
-#if DT_INST_PROP_BY_IDX(0, location_sda, 0) \
```

Comment:
Good point.  That was intentional; I didn't see an easy way to validate that within the per-instance macro.  I figured it'd be discovered pretty quickly if it happens.

It could also be verified at runtime, if necessary.

Reply:
In the serial driver we added a macro which either checks this or is empty, see https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/serial/uart_gecko.c#L450 and https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/serial/uart_gecko.c#L530

Reply:
OK, I see how that works.  I'll add that.